webgl_generator/webgl_generators/stdweb_gen.rs

```diff
@@ -0,0 +1,621 @@
+        js!(
+            return @{{canvas}}.getContext("{rendering_context}");
+        ).try_into()
+    }}
+}}
+        "#, name=name, rendering_context=rendering_context)?;
+    }
+
+    Ok(())
+}
+
+fn write_const<W>(name: &str, const_: &Const, registry: &Registry, dest: &mut W) -> io::Result<()> where W: io::Write {
+    let const_type = process_result_type(&const_.type_, registry);
+    write!(dest, r#"
+    pub const {name}: {type_} = {value};"#,
+    name=shouty_snake(name),
+    type_=const_type.type_,
+    value=const_.value
+    )?;
+    Ok(())
+}
+
+fn write_attribute<W>(name: &str, attribute: &Attribute, registry: &Registry, dest: &mut W) -> io::Result<()> where W: io::Write {
+    if attribute.getter {
+        let result_type = process_result_type(&attribute.type_, registry);
+        let expr = result_type.wrapper.wrap(
+            &format!("(js! {{ return @{{self}}.{raw_name}; }} )", raw_name=name)
+        );
+
+        write!(dest, r#"
+
+    pub fn {name}(&self) -> {type_} {{
+        {expr}
+    }}"#,
+        name=unreserve(snake(name)),
+        type_=result_type.type_,
+        expr=expr
+        )?;
+    }
+    if attribute.setter {
+        let mut gc = GenericContext::new();
+        let arg_type = process_arg_type(&attribute.type_, registry, &mut gc);
+        write!(dest, r#"
+
+    pub fn set_{name}{gargs}(&self, value: {type_}){gwhere} {{
+        js!( @{{self}}.{raw_name} = @{{{value}}}; );
+    }}"#,
+        name=snake(name),
+        raw_name=name,
+        type_=arg_type.type_,
+        gargs=gc.args(),
+        gwhere=gc.constraints(),
+        value=arg_type.wrapper.wrap("value")
+        )?;
+    }
+    Ok(())
+}
+
+fn write_get_extension<W>(dest: &mut W) -> io::Result<()> where W: io::Write {
+    write!(dest, r#"
+
+    pub fn get_extension<E: Extension>(&self) -> Option<E> {{
+        (js! {{ return @{{self}}.getExtension({{E::NAME}}); }} ).try_into().ok()
+    }}"#)
+}
+
+fn write_operation<W>(name: &str, index: usize, operation: &Operation, registry: &Registry, dest: &mut W) -> io::Result<()> where W: io::Write {
+    match name {
+        "getExtension" => return write_get_extension(dest),
+        _ => {}
+    }
+
+    let mut rust_name = unreserve(snake(name));
+    if index > 0 {
+        rust_name = format!("{}_{}", rust_name, index);
+    }
+
+    let mut gc = GenericContext::new();
+
+    struct OperationArg {
+        arg: String,
+        js_arg: String
+    }
+
+    let args: Vec<_> = operation.args.iter().map(|a| {
+        let processed = process_arg_type(&a.type_, registry, &mut gc);
+        let arg = format!("{}: {}", unreserve(snake(&a.name)), processed.type_);
+        let js_arg = format!("@{{{}}}", processed.wrapper.wrap(&unreserve(snake(&a.name))));
+        OperationArg { arg, js_arg }
+    }).collect();
+
+    let rust_args = args.iter().map(|a| a.arg.clone()).collect::<Vec<_>>().join(", ");
+    let js_args = args.iter().map(|a| a.js_arg.clone()).collect::<Vec<_>>().join(", ");
```

Comment:
itertools has [`join`](https://docs.rs/itertools/0.7.6/itertools/fn.join.html) if you wanted to use that to skip the intermediate `collect`!